### DISCUSSION TITLE
Failing to query faster than 5s

### DISCUSSION THREAD

**Discussion by Isaac-Shuman** [grafana/grafana-infinity-datasource#935] [Q&A]
The plugin is only requerying when I refresh the dashboard (or set the dashboard to auto refresh). As a test, I'm currently trying to get it to query the current time from a Flask API at a rate faster than 5s (the fastest auto refresh rate).

Here is my configuration for the Infinity panel:
![image](https://github.com/user-attachments/assets/72b7f4c0-ee4d-4848-bfbd-b202b3f95be3)

I am using Version 2.9.2 of Infinity. I am running Grafana via the docker enterprise image [https://hub.docker.com/r/grafana/grafana-enterprise](https://hub.docker.com/r/grafana/grafana-enterprise)

**Comment by stuartwaring** [comment]
I have a similar request. I want the queries to rerun automatically at a set frequency, independently of the dashboard refresh rate. Is there a way to do this?

**Comment by yesoreyeram** [comment]
Sorry. Nope the plugin don't control the refresh but grafana. So you can configure minimum refresh interval in grafana config

**Reply by yesoreyeram** [reply]
https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#min_refresh_interval

**Comment by stuartwaring** [comment]
Apologies, I know the whole dashboard can be refreshed automatically but I want individual queries to run at different times. According to this page https://grafana.com/docs/grafana/latest/dashboards/use-dashboards/ the queries should be able to be set to a schedule

> By default, Grafana does not automatically refresh the dashboard. Queries run on their own schedule according to the panel settings.

**Reply by yesoreyeram** [reply]
I am sorry. That is not possible atm. If this is important for you, kindly log a feature request in the grafana repository.

Thanks

**Reply by yosiasz** [reply]
hello @yesoreyeram, I have configured refresh rate at 5s and I am not seeing any re-query of my rest api endpoint. What do I need to do for this to work please?

**Reply by yesoreyeram** [reply]
To get refresh interval faster than 5s, you need to configure https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#min_refresh_interval. Note: this is a global settings and you can control only at dashboard level. You can't set refresh interval per panel.
